### DISCUSSION TITLE
Gracefully terminate/kill a job in Airflow

### DISCUSSION THREAD

**Discussion by BhanuuPrakash** [apache/airflow#32329] [Q&A]
I have an airflow task which calls a Golang executable file and it runs for about 2 or 3 hours. I tried to stop this DAG by clicking on Mark Success / Mark Failed. I observed that the connections that are made during the run in golang are not closed even though I have written the logic to close the connections when a signal from OS (SIGTERM / SIGKILL) comes. 

My doubt is, whenever a task/DAG is marked sucess/failed while it is already running, Will the rest of the code in the golang logic runs OR it abruptly stops without terminating gracefully.

Please suggest!

**Comment by raphaelauv** [comment]
which airflow operator are you using to launch your golang executable file ?

`BashOperator` ?

**Reply by BhanuuPrakash** [reply]
Yes, BashOperator

**Reply by potiuk** [reply]
I believe Bash does not propagate signals to their child processes. You need to write your bash in the right way to terminate processes of yours when TERM signals are received. See for example here: https://linuxconfig.org/how-to-propagate-a-signal-to-child-processes-from-a-bash-script but also many other places.

Also once you find out how to do it, it would be great if you contribute back some docs around it to Bash Operator documentation explaining what should be done if you want to do something like that, it would be great contribution back and help for the others.
